### ISSUE TITLE
[TW-313] add support for notes directly into taskwarrior

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#349]
_Tomáš Čech on 2013-11-07T12:12:27Z says:_ 

 Many people seems to be using tasknotes.sh script for attaching file with additional informations which are not suited for simple annotate.

Maintaining these files separately is an easy task for one machine, more complicated task for multiple machines (but possible thanks to DropBox or similar service), but it will be hard task for alternative client accessing TaskD server.

I believe that keeping these files externally is design insufficiency and should be corrected.

I started to work on this already as it seems to be simple task and I opened this issue for two reasons

1] to know if such change will be acceptable to you
2] to discuss implementation

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2013-11-07T12:12:27Z
Modified: 2015-03-14T17:13:17Z
```

**Comment by taskwarrior** [comment]
_Tomáš Čech on 2013-11-07T12:21:17Z says:_ 

 attaching first attempt (where CmdEdit.{cpp,h} was more than inspiration).

First question I have in mind - is really wise to keep the files inlined in database?
1] it will make parsing time longer
2] variable needs to be escaped somehow to fit the the structure (newlines, double quotes)

**Comment by taskwarrior** [comment]
_Tomáš Čech on 2013-11-07T12:21:17Z says:_ 

 Attachment 0001-add-support-for-notes-inside-the-task-warrior.patch has been added with description: first attempt

**Comment by taskwarrior** [comment]
_Tomáš Čech on 2014-01-01T16:25:56Z says:_ 

 Paul Beckingham wrote:
> I want to start with a discussion of why this belongs in core?  We have a great extension that does this without requiring core changes.  It appears to be an isolated attribute that has zero effect on a task or the other attributes.  In other words, this is a UDA.

Right. That was the main purpose. Right now the extension is working well on one machine, but when one would like to synchronize the task notes, he has to do it manually. This is the first step for me - add support for notes to the core. Next step is to add support to taskd server to provide the sync functionality in the way it should be done already from user point of view. This would open way to sync also with 3rd party clients like the Android one.

> 
> Let's ignore the patch for the moment, because we need to talk about the use case first.  Later we could discuss why the patch does nothing, stores nothing, breaks convention, doesn't work with existing extensions, has no documentation, has no unit tests and is not localized.  I assume the patch attached is just a "trial balloon", for discussion.

Thank you. That was the idea behind exactly. This patch should be more convention friendly (I'd rather check it again within some other editor as emacs often lies to me :).

So I'm taking your comment as a start point in the meantime of discussion.

Thank you for your time.

**Comment by taskwarrior** [comment]
_Renato Alves on 2014-02-19T19:39:15Z says:_ 

 Maybe I'm going in a completely different direction but, aren't "annotations" supposed to behave as notes?
If not then I might have missed something.

**Comment by taskwarrior** [comment]
_Tomáš Čech on 2014-02-19T20:12:59Z says:_ 

 It can work well as single line note. In my case I use text file with outliner or some emacs mode to store details about the task.

For example I have project for reconstruction of the flat, in that project I have task 'order doors'. In text file I keep several things - dimensions of door holes, different door types I found, contact information for different offers.

I use annotations as well as log of progress - it is very useful with the date there. So I write entries like:
'I contacted Perfect Doors Ltd. and requested catalog'
'catalog arrived, need to discuss with my wife'

There is also nicely visible what is the status (last line there).
